### PR TITLE
Fixes RTSP session id parsing issue.

### PR DESCRIPTION
**Pull request by solidprinciples**
In the case of extraneous data, it does not properly parse it - this simple fix takes care of this.

### CODE REVIEW

**Comment by cgutman** [comment]
Did you see some scenario where we got a session ID that looked like this or is this more theoretical?

**Comment by solidprinciples** [comment]
Not in the Gamestream case - but in exploring the process of reverse engineering the server, an RTSP implementation (live555) appends a timeout to the session, i.e: "DEADBEEF;timeout = 60"

The strtok solution will not change the behavior of the code if there is no ";" present either.

Without this, it would be parsed improperly in the standard RTSP case - resulting in an RTSP 454 error - session not found; due to the extra appended data.

File: src/RtspConnection.c [hunk 1]
```diff
@@ -579,13 +579,20 @@ int performRtspHandshake(void) {
         }
 
         sessionId = getOptionContent(response.options, "Session");
-        if (sessionId == NULL) {
+
+        // Strips session id in the case the 
+        // RTSP server returns excess data.
+        // (i.e: DEADBEEF;timeout=)
+        char* session;
+        session = strtok(sessionId, ";");
```
Comment (cgutman) [inline src/RtspConnection.c]: Check for `sessionId == NULL` first. We can't be sure someone hasn't called `strtok()` before and not read all available tokens.

File: src/RtspConnection.c [hunk 2]
```diff
@@ -579,13 +579,20 @@ int performRtspHandshake(void) {
         }
 
         sessionId = getOptionContent(response.options, "Session");
-        if (sessionId == NULL) {
+
+        // Strips session id in the case the 
+        // RTSP server returns excess data.
+        // (i.e: DEADBEEF;timeout=)
+        char* session;
```
Comment (cgutman) [inline src/RtspConnection.c]: Move this up with the other variable declarations in this block.

File: src/RtspConnection.c (hunk 1 above)
Comment (solidprinciples) [reply src/RtspConnection.c]: This is my first pull request - I have committed updates, I think I am doing this properly.

Follow up comments of the changes are below.

File: src/RtspConnection.c (hunk 2 above)
Comment (solidprinciples) [reply src/RtspConnection.c]: Changed to use the original sessionId as strtok modifies it anyway (still won't affect functionality if there is no ";" - strtok leaves the string alone in this case).

**Comment by cgutman** [comment]
Merged, thanks

**Pull request merged**